ethcontract-common/src/artifact/hardhat.rs

```diff
@@ -139,42 +156,158 @@ impl HardHatLoader {
     /// Add network name to the list of [`denied networks`].
     ///
     /// [`denied networks`]: #structfield.networks_deny_list
-    pub fn deny_by_name(mut self, network: impl Into<String>) -> Self {
+    pub fn deny_network_by_name(mut self, network: impl Into<String>) -> Self {
         self.networks_deny_list
             .push(NetworkEntry::ByName(network.into()));
         self
     }
 
-    /// Loads an artifact from a loaded JSON value.
-    pub fn load_from_reader(&self, v: impl Read) -> Result<Artifact, ArtifactError> {
-        self.load_artifact("<unknown>", v, from_reader, from_reader)
+    /// Add contract name to the list of [`allowed contracts`].
+    ///
+    /// [`allowed contracts`]: #structfield.contracts_allow_list
```

Comment:
Not sure I am following this notation here on the left. 

Should I be reading this semantically "as the list of allowed contracts is available as a struct field at..."?

Reply:
It's just markdown hyperlink syntax:

```md
Text [with a link].

[with a link]: http://...
```

This syntax is extensively used across rust's `std` docs (see [`std/lib.rs`] for example).

[`std/lib.rs`]: https://github.com/rust-lang/rust/blob/fdd9a071474d898556891d5146fd9e8d1a10a077/library/std/src/lib.rs#L155-L189